python/GafferArnold/ArnoldTextureBake.py

```diff
@@ -0,0 +1,495 @@
+			self["__cameraTweaks"]["tweaks"]["uv_set"] = GafferScene.TweakPlug( "uv_set", "" )
+			self["__cameraTweaks"]["tweaks"]["extend_edges"] = GafferScene.TweakPlug( "extend_edges", False )
+
+			self["__cameraTweaksFilter"] = GafferScene.PathFilter()
+			self["__cameraTweaksFilter"]["paths"].setValue( IECore.StringVectorData( [ '/camera' ] ) )
+			self["__cameraTweaks"]["filter"].setInput( self["__cameraTweaksFilter"]["out"] )
+
+
+			self["__collectScenes"] = GafferScene.CollectScenes()
+			self["__collectScenes"]["sourceRoot"].setValue( "/camera" )
+			self["__collectScenes"]["rootNameVariable"].setValue( "collect:cameraName" )
+			self["__collectScenes"]["in"].setInput( self["__cameraTweaks"]["out"] )
+
+			self["__group"] = GafferScene.Group()
+			self["__group"]["in"][0].setInput( self["__collectScenes"]["out"] )
+			self["__group"]["name"].setInput( self["cameraGroup"] )
+
+			self["__parent"]["child"].setInput( self["__group"]["out"] )
+
+			self["__collectSceneRootsExpression"] = Gaffer.Expression()
+			self["__collectSceneRootsExpression"].setExpression( inspect.cleandoc(
+				"""
+				info = parent["__chunkedBakeInfo"]
+				parent["__collectScenes"]["rootNames"] = IECore.StringVectorData( info.keys() )
+				"""
+			), "python" )
+
+			self["__cameraSetupExpression"] = Gaffer.Expression()
+			self["__cameraSetupExpression"].setExpression( inspect.cleandoc(
+				"""
+				cameraName = context["collect:cameraName"]
+				info = parent["__chunkedBakeInfo"]
+				i = info[cameraName]
+				udimOffset = i["udim"].value - 1001
+				parent["__cameraTweaks"]["tweaks"]["resolution"]["value"] = imath.V2i( i["resolution"].value )
+				parent["__cameraTweaks"]["tweaks"]["u_offset"]["value"] = -( udimOffset % 10 )
+				parent["__cameraTweaks"]["tweaks"]["v_offset"]["value"] = -( udimOffset / 10 )
+				parent["__cameraTweaks"]["tweaks"]["mesh"]["value"] = i["mesh"].value
+				parent["__cameraTweaks"]["tweaks"]["uv_set"]["value"] = parent["uvSet"] if parent["uvSet"] != "uv" else ""
+				"""
+			), "python" )
+
+			self["out"].setFlags( Gaffer.Plug.Flags.Serialisable, False )
+			self["out"].setInput( self["__parent"]["out"] )
+
+	def __init__( self, name = "ArnoldTextureBake" ) :
+
+		GafferDispatch.TaskNode.__init__( self, name )
+
+		self["in"] = GafferScene.ScenePlug()
+		self["filter"] = GafferScene.FilterPlug()
+		self["bakeDirectory"] = Gaffer.StringPlug( "bakeDirectory", defaultValue = "${project:rootDirectory}/renders/${script:name}/bake/" )
+		self["defaultFileName"] = Gaffer.StringPlug( "defaultFileName", defaultValue = "${bakeDirectory}/<AOV>/<AOV>.<UDIM>.exr" )
+		self["defaultResolution"] = Gaffer.IntPlug( "defaultResolution", defaultValue = 512 )
+		self["uvSet"] = Gaffer.StringPlug( "uvSet", defaultValue = 'uv' )
+		self["aovs"] = Gaffer.StringPlug( "aovs", defaultValue = 'beauty:RGBA' )
+		self["numTasks"] = Gaffer.IntPlug( "numTasks", defaultValue = 1 )
+		self["cleanUpExrs"] = Gaffer.BoolPlug( "cleanUpExrs", defaultValue = False )
```

Comment:
How useful is it to keep the intermediate files? It feels like maybe it's more of a debugging thing? Maybe this plug should be hidden, or at least default to on. I haven't checked, but do we also clean up the temporary text files?

Reply:
Fair point.  Switched default, and cleaned up the index files as well.